src/SFA.DAS.EmployerAccounts.Web/Extensions/SupportUserExtensions.cs

```diff
@@ -0,0 +1,217 @@
+            Logger.Debug(logReport);
+            return Task.FromResult(0);
+        }
+
+        private static Task OnMessageReceived()
+        {
+            Logger.Debug("MessageReceived");
+            return Task.FromResult(0);
+        }
+
+        private static  Task OnRedirectToIdentityProvider()
+        {
+            Logger.Debug("RedirectToIdentityProvider");
+            return Task.FromResult(0);
+        }
+
+        private static Action<IAppBuilder> SetAuthenticationContextForStaffUser()
+        {
+            return conf =>
+            {
+                conf.Run(context =>
+                {
+                    // for first iteration of this work, allow deep linking from the support console to the teams view
+                    // as this is the only action they will currently perform.
+                    var hashedAccountId = context.Request.Query.Get(HashedAccountId);
+                    var requestRedirect = string.IsNullOrEmpty(hashedAccountId) ? "/service/index" : $"/accounts/{hashedAccountId}/teams/view";
+
+                    context.Authentication.Challenge(new AuthenticationProperties
+                    {
+                        RedirectUri = requestRedirect,
+                        IsPersistent = true
+                    },
+                    Staff);
+
+                    context.Response.StatusCode = 401;
+                    return context.Response.WriteAsync(string.Empty);
+                });
+            };
+        }
+
+        private static Action<IAppBuilder> SetAuthenticationContext()
+        {
+            return conf =>
+            {
+                conf.Run(context =>
+                {
+                    context.Authentication.Challenge(new AuthenticationProperties
+                    {
+                        RedirectUri = "/service/index",
+                        IsPersistent = true
+                    },
+                    CookieAuthenticationDefaults.AuthenticationType);
+
+                    context.Response.StatusCode = 401;
+                    return context.Response.WriteAsync(string.Empty);
+                });
+            };
+        }
+    }
+
+    public class SupportConsoleAuthenticationOptions
+    {
+        public ADFSOptions AdfsOptions { get; set; }
+    }
+
+    public class ADFSOptions
+    {
+        public string Wtrealm { get; set; }
+        public string MetadataAddress { get; set; }
+        public string Wreply { get; set; }
+        public string BaseUrl { get; set; }      
+        public string Scopes { get; set; }
+        public bool UseCertificate { get; set; }
+        public ILogger Logger { get;  set; }
```

Comment:
Is logger an ADFS configurable item?

Where are Scopes and UseCertificate used when configuring ADFS?